Cesium3DTilesSelection/include/Cesium3DTilesSelection/Tile.h

```diff
@@ -556,7 +556,8 @@ class CESIUM3DTILESSELECTION_API Tile final {
    * This method should only be called when this tile's parent is already
    * loaded.
    */
-  void upsampleParent(std::vector<CesiumGeospatial::Projection>&& projections);
+  void
+  upsampleParent(const std::vector<CesiumGeospatial::Projection>&& projections);
```

Comment:
All this is fortunately part of a [single commit](https://github.com/CesiumGS/cesium-native/pull/347/commits/6a5e0d3d4886c1f4ee408954faf38cf218e96af7).

The rules seem to be subtle, I'll read a bit in https://stackoverflow.com/questions/4938875/do-rvalue-references-to-const-have-any-use and try out a little. One crucial question might be: Should it be **possible** to move-from inside that method? I'll have to wrap my head around that, and consult things like https://stackoverflow.com/questions/12095048/what-constitutes-a-valid-state-for-a-moved-from-object-in-c11 as well. I'm currently thinking about something **roughly** along the lines of

```
size_t size = someVector.size();
doSomethingWith(someVector);
T element = vector[size-1];
```
with `doSomethingWith(std::vector<T>&& v);`, 

<sub>Imagine some "as far as I know" and "to my understanding" sprinkled in here:</sub>

After the call, the vector could be in the "moved-from"-state, and ... it's not clear (for me) what that means, exactly. Using a `const&&` could prevent that.

Reply:
Well it'd have to be more like this, right? (I assumed the last line should say `someVector` too)

```
size_t size = someVector.size();
doSomethingWith(std::move(someVector));
T element = someVector[size-1];
```

And that std::move means that after that line, `someVector` is in a "valid but unspecified state". Therefore it is unspecified whether `size - 1` is a valid index, and if it's not, it's undefined behavior.

> Should it be possible to move-from inside that method?

Yes. Technically you can move from any non-const reference. But the r-value reference services two purposes:
* Tells clients to expect the object they pass in to be moved out.
* Requires a temporary or an explicitly std::move'd value to be passed in.

Reply:
I had another short look at this. 

>  I don't think it'll prevent code that attempts to move from compiling, it just won't actually result in a move operation. 

Attempting to do a `std::move(constRvalue)` will issue a [warning](https://docs.microsoft.com/en-us/cpp/code-quality/c26478), because it will actually create a copy.

So except for the ambiguity resolution or its use in return types that are mentioned in the SO answers, it only appears to serve the same purpose as `const` on normal reference parameters. For example, having a method that expects a non-const reference like this
```
void example(std::vector<CesiumGeospatial::Projection>& projections) {}

void Tile::upsampleParent(const std::vector<CesiumGeospatial::Projection>&& projections) {
  example(projections);
  ...
}
```
will not compile. Without the `const`, it will compile.

> > Should it be possible to move-from inside that method?
>
> Yes. 

Is there any reason to allow move-from or other mutating operations on the argument? In some way, I'd argue along the line of "make `const` what can be `const`, unless there is a reason do dedicatedly _not_ do this". On the other hand, in the cases that we're talking about here, the semantics are basically: "You have to `move` an object into that function, so the function is fully responsible for that object anyhow".

I think that there is no _really_ strong case for or against `const` here, so I'll leave the decision up to you (i.e. just say "keep the `const`s" or "remove the `const`s", and I'll update the PR accordingly).

Reply:
> Attempting to do a std::move(constRvalue) will issue a warning, because it will actually create a copy.

That's a C++ Core Guidelines checker warning, though, righy? Not a regular compiler warning. So it won't (currently) show up in CI, for example.

> Is there any reason to allow move-from or other mutating operations on the argument?

In this case, yes absolutely. Moving `projections` into `upsampleParent` and then into the worker lambda avoids a heap allocation. That heap allocation isn't the end of the world here, of course, but it's so easy to avoid that there's no reason not to.

So in summary:
1. Don't change any code from `Whatever&&` to `const Whatever&&` because that is weird and confusing at best and totally breaks the intention of the code at worst.
2. Don't change any code from `Whatever&&` to `const Whatever&` in this PR because we're _probably_ using move semantics for a reason and  we shouldn't change that in a PR like this with over hundred files changed. If you think move semantics are unnecessary or harmful somewhere, open it as a separate PR.

I think you may have already done this. Will double check and merge.